### PR TITLE
Revisit docs

### LINKED ISSUE #88: Create more coummunity guides
* [CONTRIBUTING](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors)
* [CODE of CONDUCT](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project)

### PR DESCRIPTION
**Pull request by jomey**
This is an effort to consolidate the different development setup instructions we had and centralize them to be all in the `/docs` folder. Also adds more content to the contributing guide and adds that to the root level of the repository.

Addresses the last item and closes #88

### CODE REVIEW

File: docs/community/how_to_contribute.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+=======================
+
+Before you submit a pull request, check that it meets these guidelines:
+
+1. The pull request should include tests.
+2. If the pull request adds functionality, the docs should be updated. Put
+   your new functionality into a function with a docstring.
+3. The pull request should work for Python 3.5, 3.6, 3.7 and 3.8, and for PyPy.
+   Check
+   ..    https://travis-ci.com/SnowEx/snowexsql/pull_requests
```
Comment (micah-prime) [inline docs/community/how_to_contribute.rst:133]: I think this should be updated to 
```suggestion
   ..    https://github.com/SnowEx/snowexsql/pulls
```

File: docs/community/how_to_contribute.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+A reminder for the maintainers on how to deploy.
+Make sure all your changes are committed.
+Then run::
+
+$ bump2version patch # possible: major / minor / patch
+$ git push
+$ git push --tags
+
+Travis will then deploy to PyPI if tests pass.
```
Comment (micah-prime) [inline docs/community/how_to_contribute.rst:148]:
```suggestion
Once the tag is merged, a release can be published with 
.. https://github.com/SnowEx/snowexsql/releases/new
The release name should follow the convention `snowexsql-v0.4.1`
```

**Review by micah-prime — COMMENTED** [review]
Looks great. I just had some minor tweaks to the `how_to_contribute.rst` file

**Review by micah-prime — APPROVED** [review]

**Comment by jomey** [comment]
The local gallery extensions is currently breaking the build on ReadTheDocs.io

Couldn't find a solution and this what I have researched so far:
```
The `local_extensions` key in the config.yaml only works when using the
'jb build book' command. Deploying on 'ReadTheDocs' however uses the
'jb config sphinx' command, which ignores that (currently). Trying a 
workaround to set the PYTHONPATH manually before executing the build
did not work.
```

**Comment by jomey** [comment]
Decided to roll back to a manual build of the gallery using the [Jupyter Book Card feature](https://jupyterbook.org/en/stable/content/components.html#cards). This not ideal, but lets us build and roll out the updates.

CC: @aaarendt

**Review by micah-prime — APPROVED** [review]
Heck yeah - nice work!

**Pull request merged**